examples/Chat/Example01_SimpleChat_Cancellations.cs

```diff
@@ -0,0 +1,30 @@
+﻿using NUnit.Framework;
+using OpenAI.Chat;
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Threading;
+
+namespace OpenAI.Examples;
+
+public partial class ChatExamples
+{
+    [Test]
+    public void Example01_SimpleChat_Cancellations()
+    {
+        ChatClient client = new(model: "gpt-4o", Environment.GetEnvironmentVariable("OPENAI_API_KEY"));
+
+        CancellationTokenSource ct = new CancellationTokenSource();
+        RequestOptions options = new() { CancellationToken = ct.Token };
+
+        // The following code will be simplified in the future.
+        var wireFormat = new ModelReaderWriterOptions("W");
+        ChatMessage message = ChatMessage.CreateUserMessage("Say 'this is a test.'");
+        BinaryData json = ModelReaderWriter.Write(message, wireFormat);
+
+        ClientResult result = client.CompleteChat(BinaryContent.Create(json), options);
```

Comment:
This `BinaryContent` contains the JSON for one message:

```json
{
    "role": "user",
    "content": "Say \u0027this is a test.\u0027"
}
```

This will result in a 400 error.

The appropriate format should be:

```json
{
    "model": "gpt-4o",
    "messages": [
        {
            "role": "user",
            "content": "Say \u0027this is a test.\u0027"
        }
    ]
}
```

Reply:
Yeah, I forgot about this. Now I get the context of your question about models not corresponding 1-1 to the payloads returned from protocol methods.